Lib/multiprocessing/shared_memory.py

```diff
@@ -540,19 +671,39 @@ def create(self, c, typeid, *args, **kwargs):
         return Server.create(self, c, typeid, *args, **kwargs)
 
     def shutdown(self, c):
+        "Call unlink() on all tracked shared memory then terminate the Server."
         self.shared_memory_context.unlink()
         return Server.shutdown(self, c)
 
+    def track_segment(self, c, segment_name):
+        "Adds the supplied shared memory block name to Server's tracker."
+        self.shared_memory_context.register_segment(segment_name)
 
-class SharedMemoryManager(SyncManager):
+    def release_segment(self, c, segment_name):
+        """Calls unlink() on the shared memory block with the supplied name
+        and removes it from the tracker instance inside the Server."""
+        self.shared_memory_context.destroy_segment(segment_name)
+
+    def list_segments(self, c):
+        """Returns a list of names of shared memory blocks that the Server
+        is currently tracking."""
+        return self.shared_memory_context.segment_names
```

Comment:
shouldn't these 3 methods be private?

Reply:
This specifically is useful as a debugging tool.  It can be invoked remotely via `multiprocessing.managers.dispatch()` and it has helped me both troubleshoot and verify things.

Reply:
This is something lacking in the documentation on mangers.  Admittedly it is difficult to do a good job explaining the uses for low-level controls sometimes but `get_server()` enables low-level control, providing the option to directly control the starting and stopping of the `Server` whereas calling `Manager().start()` asks the manager to do that for you.

Your encouragement that it's not worth documenting `SharedMemoryServer` makes sense.  I believe that we do still need to improve documentation around managers and their use of the `Server` class.